mesonbuild/mparser.py

```diff
@@ -67,7 +67,9 @@ def __init__(self, text, line, lineno, colno, start_line, start_lineno, start_co
             # Followed by a message saying where the block started.
             # Followed by the line of the block start.
             # Followed by a caret for the block start.
-            super().__init__("%s\n%s\n%s\nFor a block that started at %d,%d\n%s\n%s" % (text, line, '%s^' % (' ' * colno), start_lineno, start_colno, start_line, "%s^" % (' ' * start_colno)))
+            super().__init__("{0}\n{1}\n{2}\nFor a block that started at {3},{4}\n{5}\n{6}".format(text, line,
+                                                                                                   '{0}^'.format(' ' * colno), start_lineno, start_colno, start_line,
+                                                                                                   "{0}^".format(' ' * start_colno)))
```

Comment:
This would be much more readable if you brought the arguments to format down to a new line. I'd also prefer to get rid of the nested format calls if possible,

```python
super().__init__(longstr.format(
    text,
    line,
    ...))
```

Reply:
Made it shorter as suggested.